dao2caom2/dao_name.py

```diff
@@ -0,0 +1,203 @@
+    - support compressed and uncompressed product files
+    - uncompressed product files have an extension added to the input
+      names - lower case if there's a single input, upper case if there's
+      multiple inputs, as far as I can tell right now.
+    """
+
+    DAO_NAME_PATTERN = '*'
+
+    def __init__(self, obs_id=None, fname_on_disk=None, file_name=None,
+                 artifact_uri=None):
+        if artifact_uri is not None:
+            scheme, archive, file_name = mc.decompose_uri(artifact_uri)
+        if file_name is not None:
+            self.fname_in_ad = file_name
+            self._file_name = file_name.replace('.header', '')
+        if obs_id is None:
+            obs_id = DAOName.get_obs_id(file_name)
+            file_id = mc.StorageName.remove_extensions(file_name)
+        else:
+            file_id = obs_id
+        super(DAOName, self).__init__(
+            obs_id, COLLECTION, DAOName.DAO_NAME_PATTERN, fname_on_disk)
+        self._file_id = file_id
+        self._logger = logging.getLogger(__name__)
+        self._logger.error(self)
+
+    def __str__(self):
+        return f'obs id {self._obs_id} file name {self.file_name} ' \
+               f'file id {self._file_id}'
+
+    @property
+    def file_id(self):
+        return self._file_id
+
+    @property
+    def file_name(self):
+        return self._file_name
+
+    @property
+    def file_uri(self):
+        return f'{self.scheme}:{self.archive}/{self.file_name}'
+
+    def is_valid(self):
+        return True
+
+    @property
+    def prev(self):
+        """The preview file name for the file."""
+        return '{}_prev.jpg'.format(self.file_id)
+
+    @property
+    def product_id(self):
+        return self._file_id
+
+    @property
+    def thumb(self):
+        """The thumbnail file name for the file."""
+        return '{}_prev_256.jpg'.format(self.file_id)
+
+    @staticmethod
+    def is_derived(entry):
+        # entry is a uri
+        result = False
+        if re.match('ad:DAO/dao_[c]\\d{3}_\\d{4}_\\d{6}_[BF].\\w', entry):
+            result = True
+        return result
+
+    @staticmethod
+    def is_processed(entry):
+        # the entry is a uri
+        file_id = mc.CaomName(entry).file_id
+        result = False
+        if (re.match('dao_[cr]\\d{3}_\\d{4}_\\d{6}_[aevBF]', file_id) or
+                re.match('dao_[p]\\d{3}_\\d{6}(u|v|y|r|i|)', file_id)):
+            result = True
+        return result
+
+    @staticmethod
+    def is_processed_image(entry):
+        # entry is a uri
+        result = False
+        if re.match('ad:DAO/dao_[c]\\d{3}_\\d{4}_\\d{6}_[aBF].\\w', entry):
+            result = True
+        return result
+
+    @staticmethod
+    def is_spectrum(entry):
+        # entry is a uri
+        file_id = mc.CaomName(entry).file_id
+        result = False
+        # processed DAO spectrum
+        if (re.match('dao_[cr]\\d{3}_\\d{4}_\\d{6}_[aevBF]', file_id) or
```

Comment:
Unfortunately co-added imaging flats and biases also have file_id's like dao_c182_2019_012345_[BF].  In the original dao2caom2 I used the FITS header OBSMODE value to first see if the file is an 'Imaging' observation.  I added an obsmodes hash to allow it to be imaging or Imaging but all should be 'Imaging'.  I think you usually use 'toupper' in such cases anyway?  But, it might be better to the INSTRUME value and if "Imager" is in the string it's an image.  This is better since if I ever do a special preview for spectropolarimeter observations it will capture that info as well.  (OBSMODE = single-slit spectroscopy for regular spectroscopy and spectropolarimetry for historical reasons)

Reply:
Using 'Imaging' as obsmode throughout.

The is_spectrum function is gone now, because it's of no use when initially relying on the OBSMODE.